### PR TITLE
Feature: Updating Error Handling And Test Coverage.

### PR DESCRIPTION
**Pull request by nickfang**
## Overview
This PR implements a comprehensive refactor of error handling throughout the Vestaboard Local application, along with significant improvements to test coverage across all widget modules and core functionality.

### Error Handling Improvements
- Major refactor of [errors.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) - Expanded error handling with 191+ new lines of robust error types and handling
- Updated core modules (daemon.rs, main.rs, scheduler.rs) to use improved error handling patterns
- Enhanced error propagation throughout the widget system
### Test Coverage Expansion
- Added comprehensive test suites for previously untested components:
  - [error_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) (227 lines) - Complete error handling test coverage
  - [jokes_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) (66 lines) - Jokes widget tests
  - [sat_words_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) (171 lines) - SAT words widget tests
- Significantly expanded existing tests:
  - [widget_utils_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) - Major test improvements and additions
  - [text_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) - Enhanced text widget test coverage
  - [daemon_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) and [scheduler_tests.rs](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) - Updated for new error handling
### Widget Improvements
- Weather widget refactoring (weather.rs) - Substantial code cleanup and error handling improvements
- Widget utilities enhancement - Added format message functionality to widget_utils.rs
- Updated all widget modules to properly handle and propagate errors
- Comprehensive error handling implemented across all modules
### 🧪 Testing
- All existing functionality maintained with improved error handling
- New test files provide comprehensive coverage for error scenarios
- Widget functionality thoroughly tested with edge cases
### 🎯 Impact
- Improved reliability through better error handling and recovery
- Enhanced maintainability with comprehensive test coverage
- Better user experience through more informative error messages
- Increased code quality through refactoring and testing

### CODE REVIEW

File: src/widgets/widget_utils.rs [hunk 1]
```diff
@@ -76,6 +78,21 @@ pub fn center_message(mut message: Vec<String>, height: usize) -> WidgetOutput {
...
+            c.is_ascii_digit() ||
+            c == ' ' ||
+            "!@#$()-+&=;:'\"%,./?".contains(c) ||
+            "DROYGBVWKF".contains(c);
+
+        if !is_valid {
+            return None; // Invalid character found
+        }
+    }
+
```
Comment (nickfang) [inline src/widgets/widget_utils.rs:95]: Validation is also done in api_broker.  This should be move to somewhere that makes more sense.  Will check it in for now and refactor later.

**Review by nickfang — COMMENTED** [review]
Self-review
